pkg/nodeconfig/nodeconfig.go

```diff
@@ -162,12 +163,27 @@ func (nc *nodeConfig) Configure() error {
 		return err
 	}
 
+	tlsSecret := &core.Secret{}
+	err = nc.client.Get(context.TODO(), types.NamespacedName{Name: tls.Secret,
+		Namespace: nc.wmcoNamespace}, tlsSecret)
+	if err != nil {
+		return fmt.Errorf("unable to get secret %s: %w", nc.wmcoNamespace, err)
+	}
+	var tlsPresent bool
+	if _, ok := tlsSecret.Annotations[metadata.TlsAnnotation]; ok {
```

Comment:
@mansikulkarni96  I'm not clear where/who initially sets the  proposed `windowsmachineconfig.openshift.io/tls-update` annotation to the secret?

Reply:
It is set in secret_controller.go-> https://github.com/openshift/windows-machine-config-operator/blob/cc01700785ac78e1e47a68ea59823a48d473b767/controllers/secret_controller.go#L165